core/src/main/java/hudson/PluginManager.java

```diff
@@ -2105,7 +2105,9 @@ public String toString() {
     }
     public static boolean FAST_LOOKUP = !SystemProperties.getBoolean(PluginManager.class.getName()+".noFastLookup");
 
+    @Deprecated
     public static final Permission UPLOAD_PLUGINS = new Permission(Jenkins.PERMISSIONS, "UploadPlugins", Messages._PluginManager_UploadPluginsPermission_Description(),Jenkins.ADMINISTER,PermissionScope.JENKINS);
```

Comment:
can you change the description / title of this permission so that it shows as deprecated to users?